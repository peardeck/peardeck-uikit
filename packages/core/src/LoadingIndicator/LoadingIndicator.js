import React from "react";
import styled, { keyframes } from "styled-components";
import { IconLoading } from "@pearkit/icons";
const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;
const RotatingLoader = styled(IconLoading)`
  display: inline-block;
  animation: ${rotate} 2s linear infinite;
  padding: 2rem 1rem;
  font-size: 4rem;
`;
const StyledLoadingIndicator = styled.div`
  text-align: center;
  display: inline-block;
`;
export const LoadingIndicator = () =>
  React.createElement(
    StyledLoadingIndicator,
    null,
    React.createElement(RotatingLoader, null),
    React.createElement("div", null, "Loading...")
  );
//# sourceMappingURL=LoadingIndicator.js.map
