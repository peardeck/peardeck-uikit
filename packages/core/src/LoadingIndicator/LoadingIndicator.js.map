{"version":3,"file":"LoadingIndicator.js","sourceRoot":"","sources":["LoadingIndicator.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,MAAM,EAAE,EAAE,SAAS,EAAE,MAAM,mBAAmB,CAAC;AACtD,OAAO,EAAE,WAAW,EAAE,MAAM,gBAAgB,CAAC;AAE7C,MAAM,MAAM,GAAG,SAAS,CAAA;;;;;;;;CAQvB,CAAC;AAEF,MAAM,cAAc,GAAG,MAAM,CAAC,WAAW,CAAC,CAAA;;eAE3B,MAAM;;;CAGpB,CAAC;AAEF,MAAM,sBAAsB,GAAG,MAAM,CAAC,GAAG,CAAA;;;CAGxC,CAAC;AAEF,MAAM,CAAC,MAAM,gBAAgB,GAAG,GAAgB,EAAE,CAAC,CACjD,oBAAC,sBAAsB;IACrB,oBAAC,cAAc,OAAG;IAClB,8CAAqB,CACE,CAC1B,CAAC"}