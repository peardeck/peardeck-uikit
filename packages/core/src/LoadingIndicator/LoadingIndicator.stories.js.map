{"version":3,"file":"LoadingIndicator.stories.js","sourceRoot":"","sources":["LoadingIndicator.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,EAAE,SAAS,EAAE,MAAM,kBAAkB,CAAC;AAC7C,OAAO,EAAE,gBAAgB,EAAE,MAAM,oBAAoB,CAAC;AAEtD,SAAS,CAAC,uBAAuB,EAAE,MAAM,CAAC,CAAC,GAAG,CAC5C,SAAS,EACT,GAAgB,EAAE,CAAC,oBAAC,gBAAgB,OAAG,EACvC;IACE,KAAK,EAAE;;OAEJ;CACJ,CACF,CAAC"}