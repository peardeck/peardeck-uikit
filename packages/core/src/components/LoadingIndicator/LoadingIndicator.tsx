import React from "react";
import styled from "styled-components";
import { IconLoading } from "@peardeck-uikit/icons";
import { rotate } from "../../animations/rotate";
import { Box } from "../Box/Box";

const RotatingLoader = styled(IconLoading)`
  animation: ${rotate} 2s linear infinite;
`;

const LoadingIndicatorBox = styled(Box)`
  display: inline-block;
  text-align: center;
`;

export type LoadingIndicatorProps = { children?: React.ReactNode };

export const LoadingIndicator = ({
  children,
}: LoadingIndicatorProps): JSX.Element => (
  <LoadingIndicatorBox>
    <RotatingLoader size="4em" />
    {children && <Box>{children}</Box>}
  </LoadingIndicatorBox>
);

LoadingIndicator.defaultProps = {
  children: "Loading...",
};
