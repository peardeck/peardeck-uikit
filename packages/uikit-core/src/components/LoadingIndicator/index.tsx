export { LoadingIndicator, LoadingIndicatorProps } from "./LoadingIndicator";
