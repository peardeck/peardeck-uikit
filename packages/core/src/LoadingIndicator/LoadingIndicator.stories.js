import React from "react";
import { storiesOf } from "@storybook/react";
import { LoadingIndicator } from "./LoadingIndicator";
storiesOf("core/LoadingIndicator", module).add(
  "Default",
  () => React.createElement(LoadingIndicator, null),
  {
    notes: `
        # Loading Indicator
      `,
  }
);
//# sourceMappingURL=LoadingIndicator.stories.js.map
