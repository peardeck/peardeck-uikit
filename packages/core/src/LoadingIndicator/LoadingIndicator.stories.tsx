import React from "react";
import { storiesOf } from "@storybook/react";
import { LoadingIndicator } from "./LoadingIndicator";

storiesOf("core/LoadingIndicator", module).add(
  "Default",
  (): JSX.Element => <LoadingIndicator />,
  {
    notes: `
        # Loading Indicator
      `,
  }
);
