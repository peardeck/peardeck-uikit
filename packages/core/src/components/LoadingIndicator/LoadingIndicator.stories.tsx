import React from "react";
import { text } from "@storybook/addon-knobs";
import { LoadingIndicator } from "./LoadingIndicator";

export default {
  component: LoadingIndicator,
  title: "core|LoadingIndicator",
};

export const Default = () => (
  <LoadingIndicator>
    {text("Message", LoadingIndicator.defaultProps.children)}
  </LoadingIndicator>
);
