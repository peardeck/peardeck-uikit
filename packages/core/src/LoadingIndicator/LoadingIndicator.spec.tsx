import { create } from "react-test-renderer";
import React from "react";
import { LoadingIndicator } from "./LoadingIndicator";

describe("LoadingIndicator", () => {
  test("is a named export", () => {
    expect(LoadingIndicator).toBeDefined();
  });

  test("it matches the snapshot", () => {
    const component = create(<LoadingIndicator />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
