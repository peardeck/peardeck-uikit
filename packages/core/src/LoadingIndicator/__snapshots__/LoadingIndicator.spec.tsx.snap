// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator it matches the snapshot 1`] = `
.c1 {
  display: inline-block;
  -webkit-animation: iVXCSc 2s linear infinite;
  animation: iVXCSc 2s linear infinite;
  padding: 2rem 1rem;
  font-size: 4rem;
}

.c0 {
  text-align: center;
  display: inline-block;
}

<div
  className="c0"
>
  <svg
    aria-hidden="true"
    className="c1"
    focusable="false"
    height="1em"
    tabIndex={-1}
    viewBox="0 0 48 48"
    width="1em"
  >
    <title />
    <path
      d="M24 43C13.5 43 5 34.5 5 24S13.5 5 24 5V0C10.8 0 0 10.8 0 24s10.8 24 24 24 24-10.8 24-24h-5c0 10.5-8.5 19-19 19z"
      fill="#8D8B88"
    />
    <path
      d="M24 0v5c10.5 0 19 8.5 19 19h5C48 10.8 37.2 0 24 0z"
      fill="#C9DE55"
    />
  </svg>
  <div>
    Loading...
  </div>
</div>
`;
